src/Formats/FormatFactory.cpp

```diff
@@ -232,14 +233,18 @@ InputFormatPtr FormatFactory::getInput(
     const Settings & settings = context->getSettingsRef();
     const auto & file_segmentation_engine = getCreators(name).file_segmentation_engine;
 
+    size_t max_threads = settings.max_threads;
+    if (number_of_streams.has_value() && number_of_streams.value())
+        max_threads = std::max(max_threads/number_of_streams.value(),1UL);
```

Comment:
```suggestion
        max_threads = std::max(max_threads / number_of_streams.value(), 1UL);
```

Reply:
Looks like a new parameter `max_parsing_threads` was already added in this PR #47964, so I will remove this code and try to calculate the threads and send to `getInput` function.